src/game.cpp

```diff
@@ -1593,7 +1593,7 @@ void Game::addMoney(Cylinder* cylinder, uint64_t money, uint32_t flags /*= 0*/)
 
 		money -= currencyCoins * worth;
 		while (currencyCoins > 0) {
-			const uint16_t count = std::min<uint32_t>(ITEM_STACK_SIZE, currencyCoins);
+			const uint16_t count = std::min<uint32_t>(Item::items[it.second].stackSize, currencyCoins);
```

Comment:
I think it’s better to define a temporary variable to store the `stackSize`, then reuse it in the loop. Otherwise, what you’re doing is accessing `it` and getting a reference to the `itemtype`, then accessing the `itemtype` and obtaining a copy of the number in each iteration.